File: src/CommandLineCommands/ConfigCommands/ConfigBaseCommand.cs
using System;

abstract class ConfigBaseCommand : Command
{
    protected readonly ConfigStore _configStore;
    
    public ConfigFileScope? Scope { get; set; }

    public ConfigBaseCommand()
    {
        _configStore = ConfigStore.Instance;
    }

    protected ConfigFileScope GetConfigScope()
    {
        // If a scope was explicitly set via command line, use it
        if (Scope.HasValue)
        {
            return Scope.Value;
        }
        
        // Default to Any for config list, Local for other commands
        if (this is ConfigListCommand)
        {
            return ConfigFileScope.Any;
        }
        
        return ConfigFileScope.Local;
    }

    public override bool IsEmpty()
    {
        return false;
    }
}

File: ts/src/CommandLineCommands/ConfigCommands/ConfigBaseCommand.ts
import { ConfigStore } from '../../Configuration/ConfigStore';
import { ConfigFileScope } from '../../Configuration/ConfigFileScope';

/**
 * Base class for configuration commands.
 */
export abstract class ConfigBaseCommand {
  protected readonly _configStore: ConfigStore;
  
  public scope?: ConfigFileScope;
  public configFileName?: string;

  constructor() {
    this._configStore = ConfigStore.instance;
  }

  /**
   * Gets the command name.
   */
  abstract getCommandName(): string;

  /**
   * Executes the command.
   */
  abstract executeAsync(interactive: boolean): Promise<number>;

  /**
   * Checks if the command has all required parameters.
   */
  isEmpty(): boolean {
    return false;
  }
}